### PR TITLE
fix: inconsistant and duplication instance id values and expand payload of attribute instance

### PR DESCRIPTION
**Pull request by gpoujol**

### CODE REVIEW

**Comment by gpoujol** [comment]
> > 
> 
> @gpoujol did you notice this: [#900 (comment)](https://github.com/stellio-hub/stellio-context-broker/pull/900#issuecomment-1487226533) ?



> > 
> 
> @gpoujol did you notice this: [#900 (comment)](https://github.com/stellio-hub/stellio-context-broker/pull/900#issuecomment-1487226533) ?

no I had not seen. But that's added on its own, I didn't understand?

**Comment by bobeal** [comment]
> 

That's the usual Sonar analysis. But it is complaining about high code duplication.

**Comment by gpoujol** [comment]
> > 
> 
> That's the usual Sonar analysis. But it is complaining about high code duplication.

The duplication is the attributeInstance without the copy.

**Comment by bobeal** [comment]
can you rebase it on develop? it includes the modifications from the "modify attribute instance" feature.

File: search-service/src/main/kotlin/com/egm/stellio/search/service/SubscriptionEventListenerService.kt [hunk 1]
```diff
@@ -154,15 +154,16 @@ class SubscriptionEventListenerService(
...
+            attributeInstanceService.create(
+                AttributeInstance(
+                    temporalEntityAttribute = tea.id,
+                    timeProperty = AttributeInstance.TemporalProperty.OBSERVED_AT,
+                    time = notification.notifiedAt,
+                    value = entitiesIds,
+                    instanceId = notification.id,
+                    payload = payload
+                )
+            ).bind()
```
Comment (bobeal) [inline search-service/src/main/kotlin/com/egm/stellio/search/service/SubscriptionEventListenerService.kt:166]: you added a LOT of noise with all similar modifications in the PR. Whereas it seems nothing has changed...

File: search-service/src/main/resources/db/migration/V0_33__fix_inconsistent_insance_id_values.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+update attribute_instance
```
Comment (bobeal) [inline search-service/src/main/resources/db/migration/V0_33__fix_inconsistent_insance_id_values.sql:1]: fix the name of the migration file

File: search-service/src/main/kotlin/db/migration/V0_32__Expand_attribute_instance_payload.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    private fun updatePayloadAttributeInstanceAudit(
+        instanceId: URI,
+        expandedPayload: Map<String, Any>
+    ) {
+        logger.debug("Expand instance $instanceId")
+        jdbcTemplate.execute(
+            """
+            update attribute_instance_audit
+            set payload = $$expandedPayload$$
+            where instance_id = $$$instanceId$$
```
Comment (bobeal) [inline search-service/src/main/kotlin/db/migration/V0_32__Expand_attribute_instance_payload.kt:81]: hmm, `instance_id` is an URI, does it really require such escaping?

File: search-service/src/main/kotlin/db/migration/V0_32__Expand_attribute_instance_payload.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    private fun updatePayloadAttributeInstanceAudit(
+        instanceId: URI,
+        expandedPayload: Map<String, Any>
+    ) {
+        logger.debug("Expand instance $instanceId")
+        jdbcTemplate.execute(
+            """
+            update attribute_instance_audit
+            set payload = $$expandedPayload$$
```
Comment (bobeal) [inline search-service/src/main/kotlin/db/migration/V0_32__Expand_attribute_instance_payload.kt:80]: `payload` is a column of `jsonb` type

File: search-service/src/main/kotlin/com/egm/stellio/search/service/SubscriptionEventListenerService.kt (hunk 1 above)
Comment (gpoujol) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/SubscriptionEventListenerService.kt:166]: do you want me to remove this kind of change?

Comment (bobeal) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/SubscriptionEventListenerService.kt:166]: unless there is a good reason for doing this move?

Comment (gpoujol) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/SubscriptionEventListenerService.kt:166]: I thought it was more readable that way.

**Comment by bobeal** [comment]
> As said, split it into two branches:
> 
>     * one to fix the instance id problem (the urgent one)
> 
>     * one to expand old attributes instances

ah just noticed the other branch for the first point.

**Pull request closed without merging**